src/map/itemdb.c

```diff
@@ -2409,6 +2409,93 @@ static uint64 itemdb_unique_id(struct map_session_data *sd)
 	return ((uint64)sd->status.char_id << 32) | sd->status.uniqueitem_counter++;
 }
 
+static bool itemdb_read_libconfig_lapineddukddak(void)
+{
+	struct config_t item_lapineddukddak;
+	struct config_setting_t *it = NULL;
+	const char *config_filename = "db/item_lapineddukddak.conf"; // FIXME hardcoded name
+	int i = 0;
+	int count = 0;
+
+	if (!libconfig->load_file(&item_lapineddukddak, config_filename))
+		return false;
+
+	while ((it = libconfig->setting_get_elem(item_lapineddukddak.root, i++))) {
+		if (itemdb->read_libconfig_lapineddukddak_sub(it, config_filename))
+			++count;
+	}
+
+	libconfig->destroy(&item_lapineddukddak);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", count, config_filename);
+	return true;
+}
+
+static bool itemdb_read_libconfig_lapineddukddak_sub(struct config_setting_t *it, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+
+	struct item_data *data = NULL;
+	const char *name = config_setting_name(it);
+	const char *str = NULL;
+	int i32 = 0;
+
+	if ((data = itemdb->name2id(name)) == NULL) {
+		ShowWarning("itemdb_read_libconfig_lapineddukddak_sub: unknown item '%s', skipping..\n", name);
+		return false;
+	}
+
+	data->lapineddukddak = aCalloc(1, sizeof(struct item_lapineddukddak));
+	if (libconfig->setting_lookup_int(it, "NeedCount", &i32))
+		data->lapineddukddak->NeedCount = (int16)i32;
+
+	if (libconfig->setting_lookup_int(it, "NeedRefineMin", &i32))
+		data->lapineddukddak->NeedRefineMin = (int8)i32;
+
+	struct config_setting_t *sources = libconfig->setting_get_member(it, "SourceItems");
```

Comment:
move declaration to the top of the functions and check validity of ```libconfig->setting_get_member``` return value